tests/test_search.py

```diff
@@ -346,3 +348,17 @@ def assert_is_sorted(item, path: str):
             assert item == sorted(item), f'{err_path} is not sorted'
         for i, child in enumerate(item):
             assert_is_sorted(child, f'{path}[{i}]')
+
+
+@pytest.mark.parametrize('directory', (
+    directory for directory in (TESTS_ROOT / 'js' / 'roots').iterdir()
+))
```

Comment:
```suggestion
JAVASCRIPT_TEST_ROOTS = list((TESTS_ROOT / 'js' / 'roots').iterdir())
@pytest.mark.parametrize('directory', JAVASCRIPT_TEST_ROOTS)
```

For parametrized tests, I prefer having a non-iterator to ensure that we have everything *before* launching the test.

Reply:
Although it may be non-obvious, this should evaluate and construct a non-iterator value for `JAVASCRIPT_TEST_ROOTS` -- the `iterdir()` call is wrapped within a `list(...)` call that should evaluate it.  (or maybe I misunderstood the suggestion here?)

Reply:
Ma suggestion is to have a list in memory when calling `parametrize` and not a generator constructed on the fly. Here,

```
(directory for directory in (TESTS_ROOT / 'js' / 'roots').iterdir())
``` 

is a generator (not a tuple!) (so it is still lazily evaluated) and actually equivalent to `(TESTS_ROOT / 'js' / 'roots').iterdir()`. In general, pytest recommends to have eager values and not iterated values (for instance, if you use my suggestion, it's easy to overwrite the value of JAVASCRIPT_TEST_ROOTS).

Reply:
Argh, my mistake.  Somehow I read your code review suggestion (that _does_ include a `list(...)` call) as the current state of the test code.  However, indeed, as-is this branch does use an iterator.  Fixing this momentarily..